### PR TITLE
Remove beta headers from stripe requests

### LINKED ISSUE #4514: Stripe Link: Remove beta headers for Stripe requests
### Description
Beta headers for Stripe Link would be deprecated in the next couple of months, and we need to remove the headers introduced with issue #3843.

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #4514 

#### Changes proposed in this Pull Request
This PR removes beta headers from Stripe requests, as the beta headers are deprecated on Stripe side.

#### Testing instructions

* Create an order and pay it with Link payment method
* No errors should occur.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Review by kalessil — APPROVED** [review]
LGTM!

File: client/checkout/api/index.js [hunk 1]
```diff
@@ -37,10 +37,6 @@ export default class WCPayAPI {
 			options.betas = betas;
 		}
 
-		if ( betas.includes( 'link_beta_2' ) ) {
-			options.apiVersion = '2020-08-27;link_beta=v1';
```
Comment (mordeth) [inline client/checkout/api/index.js:41]: Do we have to remove the API version also?

**Review by mordeth — APPROVED** [review]
LGTM! Added a comment about API version.

File: client/checkout/api/index.js (hunk 1 above)
Comment (dpaun1985) [reply client/checkout/api/index.js:41]: I removed it, and it seems that it's not needed anymore. The order process works without it.

File: client/checkout/api/index.js [hunk 2]
```diff
@@ -74,10 +70,7 @@ export default class WCPayAPI {
 			if ( isUPEEnabled ) {
 				let betas = [ 'card_country_event_beta_1' ];
 				if ( isStripeLinkEnabled ) {
-					betas = betas.concat( [
-						'link_autofill_modal_beta_1',
-						'link_beta_2',
-					] );
+					betas = betas.concat( [ 'link_autofill_modal_beta_1' ] );
```
Comment (kalessil) [inline client/checkout/api/index.js:73]: Interesting, did we clarify with Stripe the need of `link_autofill_modal_beta_1`?

**Review by kalessil — APPROVED** [review]
LGTM! I left a comment, but we can clarify it async once merging this one.

File: client/checkout/api/index.js (hunk 2 above)
Comment (dpaun1985) [reply client/checkout/api/index.js:73]: > Interesting, did we clarify with Stripe the need of `link_autofill_modal_beta_1`?

I tested without this meta, and some errors were thrown when working on this issue. I will re-test it, and double check Stripe.

Comment (kalessil) [reply client/checkout/api/index.js:73]: Gotcha! Once asked Stripe (please cc me in the thread), please go ahead and merge the PR.

**Pull request merged**